hail/src/main/scala/is/hail/variant/ReferenceGenome.scala

```diff
@@ -240,6 +240,12 @@ case class ReferenceGenome(name: String, contigs: Array[String], lengths: Map[St
       else
         fatal(s"Invalid interval `$i' found. End `$end' is not within the range [1-${ contigLength(end.contig) }] for reference genome `$name'.")
     }
+
+    if (!Interval.isValid(locusType.ordering, start, end, includesStart, includesEnd))
```

Comment:
I feel like this check should be on the interval constructor in the IR, then we will check this potential bug for generic intervals

Reply:
I don't know why the check wasn't on interval constructors either. Maybe because the previous interval semantics allowed degenerate intervals, I was being extra cautious of making semantics changes.